### PR TITLE
Add Dockerfile and docker-compose.yml

### PR DESCRIPTION
**Pull request by dbirks**
Hi! Thanks for making umami. I've just started to play with it, but here's a starting point with some docker files.

To start it, you should be able to just run `docker-compose up`.

The `postgres` image has a initdb folder that you can drop any `.sql` file into, so I mounted the `sql/schema.postgresql.sql` file into there to have it run on first start up.

Some gotchas that I can think of:
- If you change some of the code and want to build a new container, you'll want to rebuild and invalidate the cache with: `docker-compose build --no-cache umami`.
- The postgres data is stored in a docker volume `postgres-data`, so if you want to delete the database and restart everything, I usually run `docker-compose down --volumes` before standing it back up.

Oh and I wasn't sure what to do for documentation, so feel free to change that (or any of this for that matter).

Linking relevant issue: https://github.com/mikecao/umami/issues/6.

### CODE REVIEW

**Comment by mikecao** [comment]
Looks good, just give me some time to test it out first.

**Comment by dbirks** [comment]
Yep no problem :+1:

**Comment by Jean-Cote** [comment]
Could you please create a [CapRover One Click App](https://github.com/caprover/one-click-apps) to make using umami easier for beginners like me?

**Comment by salmanulfarzy** [comment]
@Jean-Cote I'm actually working on it. Will be creating a PR there soon.

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    build: .
+    ports:
+      - "3000:3000"
+    environment:
+      DATABASE_URL: postgresql://umami:umami@postgres:5432/umami
+      HASH_SALT: replace-me-with-a-random-string
+  postgres:
+    image: postgres:alpine
+    ports:
+      - "5432:5432"
```
[the remark this answers is not in the document]
Comment (insuusvenerati) [reply docker-compose.yml:13]: Ports need to be open if accessing from outside the docker network. So if the app isn't deployed as part of the docker-compose then the ports will need to be available.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,12 @@
+FROM node:12.18-alpine
+
+COPY . /app
+WORKDIR /app
+
+RUN npm install \
```
Comment (tim-hub) [inline Dockerfile:6]: for better dependency management, better to use `npm ci`

**Comment by Jean-Cote** [comment]
That is awesome!
 

Gesendet: Donnerstag, 20. August 2020 um 19:02 Uhr
Von: "Salmanul Farzy" <notifications@github.com>
An: "mikecao/umami" <umami@noreply.github.com>
Cc: "Jean Cote" <JeanCote@gmx.de>, "Mention" <mention@noreply.github.com>
Betreff: Re: [mikecao/umami] Add Dockerfile and docker-compose.yml (#14)


 

@Jean-Cote I'm actually working on it. Will be creating a PR there soon.

**Pull request merged**